docs-markdown/src/helper/highlight-langs.ts

```diff
@@ -301,17 +302,17 @@ function getLanguageIdentifierCompletionItems(range: Range | undefined, isCancel
         const langs = getConfiguredLanguages();
         if (langs) {
             langs.forEach((lang) => {
+                const langId = lang.aliases[0];
+                const markdownSample = new MarkdownString();
+                markdownSample.appendText("*Output:*");
+                markdownSample.appendCodeblock("", langId);
+
                 const item = new CompletionItem(lang.language, CompletionItemKind.Value);
+                item.detail = markdownSample.value;
+                item.documentation = new MarkdownString(`Use the _${lang.language}_ language identifer (alias: _${langId}_).`);
+                item.insertText = new SnippetString(`${langId}\n$0\n\`\`\``);
                 item.sortText = lang.language;
-                const langId = item.insertText = lang.aliases[0];
-                const doc = new MarkdownString(`Use the "${lang.language}" language identifer (alias: ${item.insertText}).`);
-                doc.appendText("\nSample:\n");
-                doc.appendCodeblock(`\`\`\`${langId}`, "markdown");
-                doc.appendCodeblock("// some code...", "markdown");
-                doc.appendCodeblock("\n", "markdown");
-                doc.appendCodeblock("```", "markdown"); // This is missing from the completion item!
-                doc.appendText("");
-                item.documentation = doc;
+
```

Comment:
What happens if there is already a closing ` ``` ` code fence?

Reply:
Great question. The code doesn't attempt to handle that scenario. Should it?

Reply:
I think it's simply something to consider, I'm not certain - but I do know that other areas in this tooling set do take into account things like this to make the extensions a bit more adaptive. An example is here, where we can take into account whether the user has made a selection or not and base how to do something on that. https://github.com/microsoft/vscode-docs-authoring/blob/master/docs-markdown/src/controllers/image-controller.ts#L126